source/develop/release-management/features/network/ipv6_subnets.html.md

```diff
@@ -152,21 +305,85 @@ provision the following networking API entities, represented in yaml:
-    network: "{{ ipv6_network.id }}"
-    fixed_ips:
-      - subnet_id: "{{ ipv6_subnet.id }}"
+    name: router0
+    interfaces:
+      - "{{ ipv6_subnet.id }}"
 ~~~~~
 
+### Configuring stateless DHCPv6 on the subnet
+
+To configure stateless Dhcpv6 on the subnet, the user should provision
+the following yaml based ansible playbook snippet:
+
+~~~~~
+- name: create a network
+  os_network:
+    cloud: "{{ cloud_name }}"
+    state: present
+    name: netv6
+  register: ipv6_network
+
+- name: attach a subnet to the logical network
+  os_subnet:
+    cloud: "{{ cloud_name }}"
+    state: present
+    network_name: netv6
+    name: subnet_ipv6
+    ip_version: 6
+    cidr: bef0:1234:a890:5678::/64
+    gateway_ip: bef0:1234:a890:5678::1
+    ipv6_address_mode: dhcpv6_stateless
+  register: dhcpv6_subnet
+
+- name: create a router to send RA messages
+  os_router:
+    cloud: "{{ cloud_name }}"
+    state: present
+    name: router0
+    interfaces:
+      - "{{ dhcpv6_subnet.id }}"
+~~~~~
+
+## UI considerations
+
+The subnet ipv6_address_mode parameter should be added to the UI, since
+updates to existing subnets are not possible through the UI, and updates to the
+ipv6_address_mode parameter are not allowed, as per the
+[networking API](https://developer.openstack.org/api-ref/network/v2/?expanded=create-subnet-detail,update-subnet-detail#update-subnet). Until the UI is updated,
+the subnet should be created via ovirt-provider-ovn RESTful API or ansible.
+
+It is outside the scope of this feature to implement the UI changes that allow
+logical routers to be provisioned, and the REST API - or its ansible wrapper
+should be used instead. An ansible example is available
+[here](#configuring-stateless-dhcpv6-on-the-subnet).
+
 ## Testing
 
-Two scenarios are recommended:
-  - single IPv6 subnet
-  - multiple IPv6 subnets connected by router
+Two scenarios are recommended, each with two different configurations, leading
+to four different tests:
+  - single IPv6 subnet using dhcpv6-stateful configuration
+  - single IPv6 subnet using dhcpv6-stateless configuration
+  - multiple IPv6 subnets connected by router using dhcpv6-stateful configuration
+  - multiple IPv6 subnets connected by router using dhcpv6-stateless configuration
+
+All four scenarios on EL8 guest VMs require a router, whereas on EL7 guest VMs
+only the dhcpv6-stateless configurations require it.
+
+### Configuration mechanism used for each option
+
+|        Option Name        | dhcpv6-stateful | dhcpv6-stateless |
+|:-------------------------:|:---------------:|:----------------:|
+| DNS recursive name server |        DHCP     |      DHCP        |
+|     Domain Search List    |        DHCP     |      DHCP        |
+|   Network Prefix          |        DHCP     |      RA          |
+|   Network Prefix Length   |        RA       |      RA          |
+|            MTU            |        RA       |      RA          |
+|     Default gateway       |        RA       |      RA          |
```

Comment:
Please add an additional line about IP spoofing protection.

Reply:
I don't understand what you mean - I'm obviously missing something.

Reply:
If a port get a IPv4 address from OVN via DHCP, this port cannot use another IPv4 address and no other port in the same network can use this IPv4 address.
Maybe the same applies for dhcpv6-stateful, but not for dhcpv6-stateless?

Reply:
I do not think that's important: when the addresses are auto-generated from the RA messages, the 'client' assigns itself an address based on the network prefix + its own mac address. It's deterministic to OVN-central - e.g. it knows the ip addresses of the clients, since it knows the prefixes, and the mac addresses assigned to the ports.

In IPv4, the IP spoofing protection mechanism comes from ARP - it only 'knows' how to resolve a preset of addresses. I'm assuming that in IPv6 it'll do the same but with SEND / NDP.

Having said that, I might be over-simplifying things here. What do you think ?

Reply:
In OVN the IP spoofing protection is implemented via port_security, set in the logical switches - and we're currently only using / configuring MAC spoofing.

The type of address mode configured and the IP address family are unrelated to IP spoofing protection, and thus I'm resolving this conversation.